client/cmd.go

```diff
@@ -103,10 +103,14 @@ func ReadPersistentCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Cont
 	}
 
 	if clientCtx.Keyring == nil || flagSet.Changed(flags.FlagKeyringBackend) {
+		keyringDir, _ := flagSet.GetString(flags.FlagKeyringDir)
+		if keyringDir == "" {
+			keyringDir, _ = flagSet.GetString(flags.FlagHome)
```

Comment:
You don't need to do this. The flag `--home` is already read and set on the `Context` (L95-98).